HIP/hip-980.md

```diff
@@ -0,0 +1,114 @@
+- As a developer of mirror node, I want to be able to test new feature locally against a real life data from a public environment, so that I have better understanding of the feature behavior.
+
+## Specification
+
+The proposed enhancement involves adding a query parameter `timestamp` to the existing routes:
+
+- `/api/v1/accounts/{idOrAliasOrEvmAddress}/tokens` to return correct historical information about an account and their associated fungible tokens for a specific timestamp
+- `/api/v1/accounts/{idOrAliasOrEvmAddress}/nfts` to return correct historical information about an account and their associated nfts for a specific timestamp
+- `/api/v1/accounts/{idOrAliasOrEvmAddress}/allowances/crypto` to return correct historical information about an account's crypto allowances for a specific timestamp
+- `/api/v1/accounts/{idOrAliasOrEvmAddress}/allowances/nfts` to return correct historical information about an account's NFT allowances for a specific timestamp
+- `/api/v1/accounts/{idOrAliasOrEvmAddress}/allowances/tokens` to return correct historical information about an account's token allowances for a specific timestamp
+- `/api/v1/tokens/{tokenId}/nfts/{serialNumber}` to return correct historical information about an NFT for a specific timestamp
+
+**Timestamp Query Parameter**
+
+A new `timestamp` query parameter will be added to search for different resources. This query parameter will represent a consensus timestamp in a Unix format in `seconds.nanoseconds` format and will have an optional comparison operator. That is, it will search for results that `equal`, `not equal`, are `greater` or `less` than the timestamp itself or its seconds component.
+
+`/api/v1/accounts/{idOrAliasOrEvmAddress}/tokens?timestamp={value}`
+`/api/v1/accounts/{idOrAliasOrEvmAddress}/nfts?timestamp={value}`
+`/api/v1/accounts/{idOrAliasOrEvmAddress}/allowances/crypto?timestamp={value}`
+`/api/v1/accounts/{idOrAliasOrEvmAddress}/allowances/nfts?timestamp={value}`
+`/api/v1/accounts/{idOrAliasOrEvmAddress}/allowances/tokens?timestamp={value}`
+`/api/v1/tokens/{tokenId}/nfts/{serialNumber}?timestamp={value}`
+
+**Request**
+```
+/api/v1/accounts/{idOrAliasOrEvmAddress}/tokens?timestamp={value}
+```
+**Example Request**
+
+```
+GET api/v1/accounts/4408244/tokens?order=desc?timestamp=1705334449.803393003
+```
+Note: if ```limit``` is not specified in the request, response will be limited to 25 by default
+
+**Response**
+
+```json
+{
+  "tokens": [
+    {
+      "automatic_association": true,
+      "balance": 1,
+      "created_timestamp": "1705334449.803393003",
+      "decimals": 0,
+      "token_id": "0.0.4328145",
+      "freeze_status": "NOT_APPLICABLE",
+      "kyc_status": "NOT_APPLICABLE"
+    }
+  ],
+  "links": {
+    "next": null
+  }
+}
+```
+
+## **Backwards Compatibility**
+
+The change is backwards compatible as it is simply adding а query parameter to an existing route. The data required to filter by timestamp is already stored in the mirror node.
```

Comment:
Add that if the timestamp parameter is omitted it will revert to the previous behavior of querying for the latest state.

Reply:
Added